.github/workflows/pull_request_common.yml

```
@@ -27,6 +27,46 @@ jobs:
               repo: context.repo.repo,
               labels: ["patch"]
             })
+  check-dependabot-pr:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.action == 'opened' && github.actor == 'dependabot[bot]' }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Node setup
+        uses: ./.github/actions/node-setup
+
+      - name: Get @vkontakte/vkui dependencies
+        id: dependencies
+        run: echo "value=$(node -p "Object.keys(require('./packages/vkui/package.json').dependencies)")" >> $GITHUB_OUTPUT
+
+      - name: Fetch Dependabot metadata
+        id: metadata
+        uses: dependabot/fetch-metadata@v2
+        with:
+          github-token: ${{ secrets.DEVTOOLS_GITHUB_TOKEN }}
+
+      - name: Check if PR updates relevant dependencies
+        id: check_deps
+        run: |
+          RELEVANT_DEPS=(${{ steps.dependencies.outputs.value }})
+          UPDATE_NEEDED=false
+
+          for DEP in "${RELEVANT_DEPS[@]}"; do
+            if [[ "${{ steps.metadata.outputs.dependency-names }}" == *"$DEP"* ]]; then
+              UPDATE_NEEDED=true
+              break
+            fi
+          done
+
+          echo "update_needed=$UPDATE_NEEDED" >> $GITHUB_OUTPUT
```

```suggestion

          # ```sh
          # @npmcli/template-oss, @octokit/rest
          # ```
          # =>
          # ```sh
          # @npmcli/template-oss @octokit/rest
          # ```
          UPDATED_DEPS=$(echo ${{ steps.metadata.outputs.dependency-names }} | sed "s/,/ /g")
          RELEVANT_DEPS=${{ steps.dependencies.outputs.value }}

          for UPDATED_DEP in $UPDATED_DEPS; do
            if [[ ${RELEVANT_DEPS[@]} =~ $UPDATED_DEP ]]; then
              echo "update_needed=true" >> $GITHUB_OUTPUT
              exit 0
            fi
          done
```

тут также прогнал через [act](https://github.com/nektos/act) и внёс корректировки

+ убрал `UPDATE_NEEDED`, т.к. нам нужно подобие `Array.prototype.some()` - выходим из цикла как-то только одно удовлетворение условию